index.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const Joi = require('joi');
+const emailer = require('./email');
+
+const DESCRIPTION_MAP = {
+    SUCCESS: 'Everything looks good!',
+    FAILURE: 'Did not work as expected.',
+    ABORTED: 'Aborted mid-flight',
+    RUNNING: 'Testing your code...',
+    QUEUED: 'Looking for a place to park...'
+};
+const DEFAULT_STATUSES = ['FAILURE'];
+// Joi Schema Validation
+const SCHEMA_ADDRESS = Joi.string().email();
+const SCHEMA_ADDRESSES = Joi.array()
+    .items(SCHEMA_ADDRESS)
+    .min(0);
+const SCHEMA_STATUS = Joi.string().allow(Object.keys(DESCRIPTION_MAP));
+const SCHEMA_STATUSES = Joi.array()
+    .items(SCHEMA_STATUS)
+    .min(0);
+const SCHEMA_EMAIL = Joi.alternatives().try(
+    Joi.object().keys({ addresses: SCHEMA_ADDRESSES, statuses: SCHEMA_STATUSES }),
+    SCHEMA_ADDRESS, SCHEMA_ADDRESSES
+    );
+const SCHEMA_BUILD_SETTINGS = Joi.object()
+    .keys({
+        email: SCHEMA_EMAIL
+    }).unknown(true);
+const SCHEMA_BUILD_DATA = Joi.object()
+    .keys({
+        settings: SCHEMA_BUILD_SETTINGS,
+        status: SCHEMA_STATUS,
+        pipelineName: Joi.string(),
+        jobName: Joi.string(),
+        buildId: Joi.number().integer(),
+        buildLink: Joi.string()
+    })
+    .required();
```

Comment:
Maybe we should make `settings` and `status` required, and the rest optional. Open to suggestions.